att_err_mon.py

```diff
@@ -1,158 +1,150 @@
-    aoacimss = m_filter_telem['data'][0]
-    dwells = events.dwells.filter(start=start, stop=stop)
+    manvrs = events.manvrs.filter(start=start, stop=stop, n_dwell__gte=1)
     obs_data = []
-    for d in dwells:
+    for m in manvrs:
         obs = {}
         try:
-            obsid = d.get_obsid()
+            obsid = m.get_obsid()
         except ValueError:
             obsid = 0
         if obsid is None:
             obsid = 0
-        #print obsid, d.start
-        # for data earlier than the existence of the AOACIMSS msid,
-        # the value can be assumed to be 'ENAB'
-        if d.manvr.kalman_start < '2015:252':
-            obs['gui_ms'] = 'ENAB'
-        elif len(aoacimss['times']) and (d.tstop > aoacimss['times'][-1]):
-            continue
-        else:
-            mid_kalman = ((DateTime(d.manvr.kalman_start).secs
-                            + DateTime(d.stop).secs) / 2)
-            kal_idx = (np.searchsorted(aoacimss['times'],
-                                        mid_kalman))
-            obs['gui_ms'] = aoacimss['values'][kal_idx]
-
-        # if the dwell isn't for the type of data we're looking for, just continue
-        if obs['gui_ms'] != msf:
-            continue
 
-        # Get one shot and maneuver to for *next* obsid
-        n = d.get_next()
-        if not n:
-            continue
-        try:
-            obs['next_obsid'] = n.get_obsid()
-        except ValueError:
-            obs['next_obsid'] = 0
+        # If there is no next maneuver, we're done as there's no interval percentiles
+        if m.get_next() is False:
+            break
 
-        obs['date'] = d.manvr.kalman_start
-        obs['datestop'] = d.stop
-        obs['time'] = DateTime(d.manvr.kalman_start).secs
-        obs['timestop'] = d.tstop
-
-        obs['manvr_angle'] = n.manvr.angle
-        obs['one_shot'] = n.manvr.one_shot
         obs['obsid'] = obsid
+        obs['date'] = m.start
+        obs['time'] = DateTime(m.start).secs
+        obs['manvr_angle'] = m.angle
+        obs['one_shot'] = m.one_shot
+        obs['one_shot_yaw'] = m.one_shot_yaw
+        obs['one_shot_pitch'] = m.one_shot_pitch
+        obs['dwell_duration'] = DateTime(m.get_next().start).secs - DateTime(m.npnt_start).secs
 
         all_err = {}
         for err_name, err_msid in zip(['roll_err', 'pitch_err', 'yaw_err'],
-                                        ['AOATTER1', 'AOATTER2', 'AOATTER3']):
-            err = fetch.Msid(err_msid, d.tstart + 500, d.stop)
+                                      ['AOATTER1', 'AOATTER2', 'AOATTER3']):
+            err = fetch.Msid(err_msid,
+                             DateTime(m.npnt_start).secs + 500,
+                             m.get_next().start)
             events.dumps.interval_pad = (0, 300)
             err.remove_intervals(events.dumps)
             events.tsc_moves.interval_pad = (0, 300)
             err.remove_intervals(events.tsc_moves)
+            err.remove_intervals(events.dark_cal_replicas)
```

Comment:
I should cut this; I think the attitude errors aren't updated during the NMAN intervals of the SRDCs anyway, so the filter here doesn't buy us anything.  Do we want to try to trim *around* those intervals?  I don't think we care.

Reply:
👍